hpx/util/tuple.hpp

```diff
@@ -303,6 +336,18 @@ namespace hpx { namespace util
                         I, typename detail::at_index<I, Ts...>::type
                     > const&>(*this).value();
             }
+
+            HPX_HOST_DEVICE
+            operator std::tuple<Ts...>() const &
+            {
+                return std::make_tuple(get<Is>()...);
```

Comment:
`std::make_tuple` decays and unwraps, it will yield the wrong `std::tuple` specialization.

Reply:
So returning `std::tuple<Ts...>(get<Is>()...)` it is.